src/qtgui/bookmarks.cpp

```diff
@@ -158,21 +158,20 @@ bool Bookmarks::save()
         stream << QString("# Tag name").leftJustified(20) + "; " +
                   QString(" color") << '\n';
 
-        QSet<TagInfo*> usedTags;
+        QMap<QString, TagInfo::sptr> usedTags;
```

Comment:
Using raw pointers here makes Gqrx larger and makes code more complicated. You may try it yourself...

Reply:
To make sure we understand each other, the proposal was to keep the old code (for now) except the necessary changes to make it work with shared pointer, eg. the only change necessary would be somethink like:

```
             BookmarkInfo& info = m_BookmarkList[iBookmark];
             for(int iTag = 0; iTag < info.tags.size(); ++iTag)
             {
-              TagInfo& tag = *info.tags[iTag];
-              usedTags.insert(&tag);
+              usedTags.insert(info.tags[iTag].get());
             }
         }
```

Reply:
I can't guarantee, that QSet<TagInfo*> will work correctly here (collect tags with unique names). It may be better to change `Bookmarks::m_TagList` type form `QList<TagInfo>` to `QMap<QString,TagInfo::sptr>` and get rid of this loop.

Reply:
Yep, the non-uniqueness of tags should be solved too, but it seems to me this tries to solve more thinks in one MR. Mey-be separate commit if it is necessary to fix #1143 or separate MR if it can be done independently would help me.

Reply:
Switching to a better container may be a different PR.
I do not know, how other people are using tags. Tag management should be improved if someone is using thousands of tags (adding a quick filter to help finding tags/bookmarks may help in this case).